primitives/src/crh/poseidon/mod.rs

```diff
@@ -940,4 +799,87 @@ mod test {
 
     }
 
+    #[test]
+    fn test_merkle_tree() {
```

Comment:
Again, please use the bench framework. An effective test, according to me, would be generating a Merkle Tree with the "normal" implementation we have in ginger-lib at primitives/merkle_tree/field_based_mht, generating a Merkle Tree with this implementation (using the same leaves of course) and checking that the Merkle Root are equals. Please remember that for the Merkle Tree at primitives/merkle_tree/field_based_mht you have to pass a number of leaves exactly equal to 2^(HEIGHT -1) for this test to be effective.

Reply:
Done.